### ISSUE TITLE
XLA CPU: lax.reduce with int16 scalar crashes runtime

### ISSUE THREAD

**Issue by jakevdp** [google/jax#4971]
Minimal repro:
```python
from jax import lax, jit
import jax.numpy as jnp

dtype = jnp.int16

def f():
  return lax.reduce(dtype(0), dtype(0), lax.max, ())
jit(f)()
```
Output (before crashing the Python runtime):
```
2020-11-19 15:24:48.038635: F external/org_tensorflow/tensorflow/compiler/xla/service/hlo_evaluator_typed_visitor.h:1816] HandleMap: unhandled primitive type for input operand: S16
Abort trap: 6
```
This error only occurs on CPU, only occurs for `int16` (not `int8` or `int32`), and only if the first argument is a scalar.

**Comment by hawkinsp** [comment]
This is actually in the HLO evaluator (i.e., when constant folding), so it will probably reproduce across all backends.

**Comment by majnemer** [comment]
This should be fixed in XLA commit tensorflow/tensorflow@9367ca6.
